### ISSUE TITLE
The asset model is breaking asset helpers

### ISSUE THREAD

**Issue by dbaines** [katalyst/koi#360]
- Start Date: 2018-12-12
- Owner: JS
- RFC PR: (leave this empty)

# Summary

We have a model named `Asset`.  
This is used for the asset manager.  
`Document` and `Image` extend off this model.  

If the main application has a route, eg:

```
resources :assets, only: [:show]
```

This will break the rails image asset helpers such as `asset_path` and `asset_url`

# Motivation

Sometimes we need `asset_path` or `asset_url` to be the rails helpers rather than the route helpers

# Drawbacks

* Large breaking change
* Asset is a nice name for things in the asset manager
* Possibly only necessary if we ever have assets on the front-end

# Alternatives

* Some sort of front-end only alias instead, eg. `koi_asset` or something
* Hard-and-fast ruling that we only allow front-end routes for documents and images and not assets. This matches Koi functionality in that there is only an "image" asset manager and a "document" asset manager and no "general" asset manager.

**Comment by Obversity** [comment]
I'm in favour of either of the alternatives. We may just be able to remove the route without breaking anything—we'll find out shorty with Tonkin I imagine. And if we do end up needing a generic route for both images and assets, I'm more than fine with it being `koi_asset_path`. In other words, I think we'd just need to change it to

```ruby
resources :assets, only: [:show], as: :koi_assets
```

**Comment by dbaines** [comment]
@Obversity Agreed. Added that, gave it a test and it looks like it's working nicely!
Fixed in dfd4ddd488b5082886bc9dc674cc4cb311f3e742
